docs/tutorial/tailable-cursor.txt

```diff
@@ -84,62 +92,101 @@ The IteratorIterator equivalent of the above :code:`foreach` follows.
       $iterator->next();
    }
 
-Note that calling :code:`$iterator->next()` again will result in an exception since
-we've exhausted :code:`$cursor`.
+.. note::
+
+   Calling ``$iterator->next()`` after the ``while`` loop naturally ends would
+   throw an exception, since all results on the cursor have been exhausted.
+
+The purpose of this example is simply to demonstrate the functional equivalence
+between ``foreach`` and manual iteration with PHP's :php:`Iterator <iterator>`
+API. For normal cursors, there is little reason to use IteratorIterator instead
+of a concise ``foreach`` loop.
 
 Wrapping a Tailable Cursor
----------------------------------
+--------------------------
 
-The following example wraps a tailable cursor in an IteratorIterator.
+In order to demonstrate a tailable cursor in action, we'll need two scripts: a
+"producer" and a "consumer". The producer script will create a new capped
+collection using :phpmethod:`MongoDB\\Database::createCollection()` and proceed
+to insert a new document into that collection each second.
 
 .. code-block:: php
 
    <?php
 
-   $manager = new MongoDB\Driver\Manager(STANDALONE);
-   $manager->executeCommand(DATABASE_NAME, new MongoDB\Driver\Command([
-      'create' => COLLECTION_NAME,
-      'capped' => true,
-      'size' => 1048576,
-   ]));
-
-   $bulkWrite = new MongoDB\Driver\BulkWrite;
-   $bulkWrite->insert(['_id' => 1]);
-   $manager->executeBulkWrite(NS, $bulkWrite);
-   $cursor = $manager->executeQuery(NS, new MongoDB\Driver\Query([], [
-      'tailable' => true,
-      'awaitData' => true,
-      'maxAwaitTimeMS' => 10,
-   ]));
-
-   foreach (new IteratorIterator($cursor) as $document) {
-      var_dump($document);
+   $database = (new MongoDB\Client)->test;
+
+   $database->createCollection('capped', [
+       'capped' => true,
+       'size' => 16777216,
+   ]);
+
+   $collection = $database->selectCollection('capped');
+
+   while (true) {
+       $collection->insertOne(['createdAt' => new MongoDB\BSON\UTCDateTime()]);
+       sleep(1);
```

Comment:
Main changes here are using the library code instead of extension API and having separate producer/consumer scripts. The only other way to reliably demonstrate tailable cursors is by periodically inserting documents during iteration (as we do in the PHPC tests), but that's quite confusing and only necessary if keeping things in one file is essential.